setup.py

```diff
@@ -0,0 +1,181 @@
+                              'later.'.format(cmd[0]))
+            return (p.returncode, stdout, stderr)
+        elif p.returncode != 0:
+            if show_warning:
+                warnings.warn('Git failed while determining revision '
+                              'count: {0}'.format(_decode_stdio(stderr)))
+            return (p.returncode, stdout, stderr)
+
+        return p.returncode, stdout, stderr
+
+    returncode, stdout, stderr = run_git(cmd)
+
+    if not sha and returncode == 128:
+        # git returns 128 if the command is not run from within a git
+        # repository tree. In this case, a warning is produced above but we
+        # return the default dev version of '0'.
+        return '0'
+    elif not sha and returncode == 129:
+        # git returns 129 if a command option failed to parse; in
+        # particular this could happen in git versions older than 1.7.2
+        # where the --count option is not supported
+        # Also use --abbrev-commit and --abbrev=0 to display the minimum
+        # number of characters needed per-commit (rather than the full hash)
+        cmd = ['rev-list', '--abbrev-commit', '--abbrev=0', 'HEAD']
+        returncode, stdout, stderr = run_git(cmd)
+        # Fall back on the old method of getting all revisions and counting
+        # the lines
+        if returncode == 0:
+            return str(stdout.count(b'\n'))
+        else:
+            return ''
+    elif sha:
+        return _decode_stdio(stdout)[:40]
+    else:
+        return _decode_stdio(stdout).strip()
+
+
+# Need a recursive glob to find all package data files if there are
+# subdirectories
+import fnmatch
+
+
+def recursive_glob(basedir, pattern):
+    matches = []
+    for root, dirnames, filenames in os.walk(basedir):
+        for filename in fnmatch.filter(filenames, pattern):
+            matches.append(os.path.join(root, filename))
+    return matches
+
+
+# Set affiliated package-specific settings
+PACKAGENAME = 'skyportal_models'
+DESCRIPTION = 'Collaborative Platform for Time-Domain Astronomy'
+LONG_DESCRIPTION = ''
+AUTHOR = 'The Skyportal Developers'
+AUTHOR_EMAIL = ''
+LICENSE = 'BSD'
+URL = 'skyportal.io'
+
+# VERSION should be PEP386 compatible (http://www.python.org/dev/peps/pep-0386)
+VERSION = __version__
+
+# Indicates if this version is a release version
+RELEASE = 'dev' not in VERSION
+
+if not RELEASE:
+    VERSION += get_git_devstr(sha=False)
+
+setup(name=PACKAGENAME,
+      version=VERSION,
+      description=DESCRIPTION,
+      py_modules=['skyportal.models', 'skyportal.schema',
+                  'skyportal.phot_enum',
+                  'baselayer.__init__',
+                  'baselayer.app.__init__',
+                  'baselayer.app.models',
+                  'baselayer.app.json_util',
+                  'baselayer.app.custom_exceptions'],
+      setup_requires=['Cython>=0.29'],
+      install_requires=[
+          'numpy>=1.18.1',
```

Comment:
Can these be parsed from requirements.txt so we don't have to update package versions in more than one place?

Reply:
These are the minimal requirements to run the package, slightly less than the full requirements.txt

Reply:
These minimum version numbers could become out of date if we do major version upgrades, right?

Reply:
https://packaging.python.org/discussions/install-requires-vs-requirements/

Reply:
@

> These minimum version numbers could become out of date if we do major version upgrades, right?

yes, I suppose they could

Reply:
I'm not sure if that answers my question -- are you saying there's no way to avoid having to update this here manually if, say, we upgrade to some later version of a package with breaking changes and update our code base such that the above minimum requirement is actually wrong, unless we manually update it?

Reply:
e.g. some major version upgrades involve breaking API changes that are not backwards-compatible

Reply:
Sorry I sent the link before your response, we got our threads crossed. Yes I think that's correct. The idea though is that only a small number of the packages in requirements.txt are acutally used here though.

Reply:
What I'm getting at is: as we tend to specify minimum versions in requirements.txt, can't we just pull the relevant minimum versions from there?

Reply:
that can be done. Will add.